### PR TITLE
Internationalization

### PR DESCRIPTION
**Pull request by emlys**

### CODE REVIEW

File: README.md
Comment (dcdenu4) [inline README.md:51]: Cool!

File: README.md
Comment (dcdenu4) [inline README.md:64]: Should this be filled in or is it leftover?

File: source/conf.py
Comment (dcdenu4) [inline source/conf.py:19]: Do we need `sphinx_intl` here as well?

File: source/locales/es/LC_MESSAGES/carbon_edge.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/carbon_edge.po:6]: Out of curiosity, what does the `, fuzzy` indicate?

File: source/locales/es/LC_MESSAGES/the_need_for.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/the_need_for.po:175]: If a translator goes to update the `msgstr` will they be able to enter escapable (?) quotes like this?

File: source/locales/es/LC_MESSAGES/the_need_for.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/the_need_for.po:330]: Out of curiosity, what's the `, python-format` here?

File: source/conf.py
Comment (emlys) [reply source/conf.py:19]: No, `sphinx_intl` isn't a sphinx extension, just a command line tool.

**Comment by emlys** [comment]
@dcdenu4, I finished the README and added a bit of functionality that was missing (to translate the handful of strings that live in `investspec.py`).

File: extensions/investspec/investspec.py
Comment (dcdenu4) [inline extensions/investspec/investspec.py:16]: Ah, interesting. Certainly a lot to keep track of!

File: extensions/investspec/investspec.py
Comment (dcdenu4) [inline extensions/investspec/investspec.py:31]: I have a stupid question, are these types defined somewhere on the `natcap.invest` side? I looked around but didn't find a similarly valid types structure... I was mostly curious because I was wondering if we could take advantage of a single source of truth on the `natcap.invest` side in case types were added in the future. Not that'd we really be messing with these types all that much. Sorry about this ramble.

File: extensions/investspec/internationalization/messages.pot
Comment (dcdenu4) [inline extensions/investspec/internationalization/messages.pot:5]: Do we want both this file, `messages.pot` and the above file `messages.po`? I think I read in one of your Readmes that we can only need to check one in because the other is easily derived? Maybe it's different for this case? Sorry if I'm not fully understanding or confusing something!

File: source/locales/es/LC_MESSAGES/carbonstorage.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/carbonstorage.po:413]: Interesting to see how tables get broken out.

File: source/locales/es/LC_MESSAGES/coastal_vulnerability.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/coastal_vulnerability.po:171]: Oh cool, it's smart enough to cut down on redundancy.

File: source/locales/es/LC_MESSAGES/contents.po
Comment (dcdenu4) [inline source/locales/es/LC_MESSAGES/contents.po:4]: Is it possible to skip `contents.rst`? I don't think there's anything to translate.

File: extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po
Comment (dcdenu4) [inline extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po:13]: I don't think I saw this key in the other PO files, should we update the other PO files for consistency?

**Review by dcdenu4 — CHANGES_REQUESTED** [review]
Hey @emlys , this is really cool and was a great way to learn more about how translations is working across InVEST. I had a few questions / comments and one general question below.

Are we committing PO files for Spanish as demonstration of what would eventually be done and as a way to test the functionality of this PR?

File: source/locales/es/LC_MESSAGES/carbon_edge.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/carbon_edge.po:6]: It's put there by the tool that generates the PO file. [This post](https://stackoverflow.com/questions/15653093/what-does-fuzzy-mean-at-the-top-of-a-xgettext-generated-pot-file) suggests it's to indicate that the header needs to be filled in. I think that's okay for now since we haven't filled in the files yet.

File: extensions/investspec/investspec.py
Comment (emlys) [reply extensions/investspec/investspec.py:31]: I don't believe there is... it would make sense to import them from `natcap.invest`. `natcap.invest` itself has no use for _translations_ of the type names, but since they're used here and would be used in the workbench, we could centralize them there.

File: extensions/investspec/internationalization/messages.pot
Comment (emlys) [reply extensions/investspec/internationalization/messages.pot:5]: I think you're thinking of the `messages.mo` that we don't need to include... those are binary "compiled" versions of the PO files. The POT and PO files should both be checked in because the POT file is the source of truth for the current set of translated strings, while the PO files store the translations for each language.

File: extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po
Comment (emlys) [reply extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po:13]: I think it's because this PO file is created by pybabel while the others are created by sphinx-intl. I guess they have slightly different headers (all the header fields are optional according to the GNU gettext standard). I don't feel it really matters since the language is defined in the path but can add it to the other PO files if you prefer

File: source/locales/es/LC_MESSAGES/contents.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/contents.po:4]: `sphinx-intl` doesn't seem to have any option to skip a file - it makes one PO file for each RST file in a given directory. And this file will get re-created whenever we do `sphinx-intl update`. We could delete it now and each future time, but I think it's also okay to keep it here. No strong opinion.

File: source/locales/es/LC_MESSAGES/carbonstorage.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/carbonstorage.po:413]: Maybe I should go through and remove the tables from the PO files... `c_dead` for instance should not be translated.

File: extensions/investspec/investspec.py
Comment (emlys) [reply extensions/investspec/investspec.py:16]: Oh, yeah. I wish there weren't so many separate translation mechanisms going on! Maybe a simpler way will become clear once we have used it a bit.

File: source/locales/es/LC_MESSAGES/the_need_for.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/the_need_for.po:330]: Oh interesting, I hadn't noticed that. The comment lines starting with `#,` are flags which have special meaning to gettext programs. The `python-format` flag indicates that the message is a python format string (clearly incorrect here). I suspect it's because of the `%` in the string. Though, it's strange that the sphinx gettext utility isn't correcting for that.

File: extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po
Comment (dcdenu4) [reply extensions/investspec/internationalization/locales/es/LC_MESSAGES/messages.po:13]: I don't think I'm familiar enough to have a preference! But I think you're right, language is embedded in the path so not necessary.

File: source/locales/es/LC_MESSAGES/contents.po
Comment (dcdenu4) [reply source/locales/es/LC_MESSAGES/contents.po:4]: I think it's fine to leave as is.

File: source/locales/es/LC_MESSAGES/carbonstorage.po
Comment (dcdenu4) [reply source/locales/es/LC_MESSAGES/carbonstorage.po:413]: I wish `sphinx-intl` had a configuration to allow for ignoring certain RST syntax, like tables. Removing table sections from the PO files each time they need to be updated seems fairly onerous. A quick google search didn't bring up anything related unfortunately.

File: extensions/investspec/investspec.py
Comment (dcdenu4) [reply extensions/investspec/investspec.py:31]: Okay, it's fine to leave as is for this PR then. I think it's fine to wait until this actually pops up as an issue before centralizing things in `natcap.invest`.

File: source/locales/es/LC_MESSAGES/carbonstorage.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/carbonstorage.po:413]: I wonder if it would be easier not to use `gettext` at all for the RST docs and try to get them translated directly by someone who's familiar with RST syntax.

File: source/locales/es/LC_MESSAGES/the_need_for.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/the_need_for.po:330]: Pretty sure this is a bug in the sphinx gettext builder.

File: source/locales/es/LC_MESSAGES/carbonstorage.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/carbonstorage.po:413]: I made a ticket to revisit this in the future: #54

File: source/locales/es/LC_MESSAGES/the_need_for.po
Comment (emlys) [reply source/locales/es/LC_MESSAGES/the_need_for.po:175]: Normal quotes can be escaped with a backslash: `\"`. These fancy quotes `“”` are used in the source RST. Is that what you meant?

**Comment by emlys** [comment]
@dcdenu4, since we're not going to get translations done before the release, I don't think the PO files should be committed. It doesn't really matter here, but I'm considering the contents of `source/locales/` to be the single source of truth for which languages are supported. I removed all PO files in 04bbadc.

**Comment by emlys** [comment]
This depends on the changes in natcap/invest#645 so let's merge this after that PR goes through.

**Review by dcdenu4 — COMMENTED** [review]
Hey @emlys, thanks for the update and creating that well defined issue about using gettext here vs translating the RST directly. I agree with removing the PO files for now until we actually have a translation case, but it was great to see how it would work and what they'd look like!

Looks like there are some conflicting files and seeing how we're waiting on another PR I'll hold off on approving so we remember to wait for that other PR and handle the conflicts.

**Comment by emlys** [comment]
@dcdenu4 natcap/invest#568 was merged, so this is ready now!

**Review by dcdenu4 — APPROVED** [review]
Thanks for all of this work and thought @emlys!

**Pull request merged**
